src/v/rpc/rpc_server.cc

```diff
@@ -104,7 +104,16 @@ rpc_server::send_reply(ss::lw_shared_ptr<server_context_impl> ctx, netbuf buf) {
 
 ss::future<> rpc_server::send_reply_skip_payload(
   ss::lw_shared_ptr<server_context_impl> ctx, netbuf buf) {
-    co_await ctx->conn->input().skip(ctx->get_header().payload_size);
+    try {
+        // If the connection is closed then this can throw an exception.
+        // In that case we want to catch and forward it to avoid a broken
+        // promise.
+        co_await ctx->conn->input().skip(ctx->get_header().payload_size);
```

Comment:
i think it might make sense to handle this back in rpc_server::dispatch_method_once ?

Reply:
I made a comment here https://github.com/redpanda-data/redpanda/pull/8519/files#r1091520039 about why I'm handling it here. Basically if I handle it in `rpc_server::dispatch_method_once` with a `finally` or similar I can only set the promise after the reply is sent or an exception is thrown. Which is why I moved the handling here. Happy to move the handling to `rpc_server::dispatch_method_once` if need be though.